internal/gcs/bridge.go

```diff
@@ -0,0 +1,356 @@
+package gcs
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/binary"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/Microsoft/hcsshim/internal/guid"
+	"github.com/sirupsen/logrus"
+)
+
+type requestMessage interface {
+	SetActivityID(id guid.GUID)
+}
+
+type responseMessage interface {
+	Err() error
+}
+
+// rpc represents an outstanding rpc request to the guest
+type rpc struct {
+	proc rpcProc
+	id   int64
+	req  requestMessage
+	resp responseMessage
+	err  error
```

Comment:
Based on the code it looks like this is the `local` error and `resp.Err` is the remote error?

Reply:
Right. I can make this clearer.

Reply:
I renamed err to brdgErr.